core/embed/rust/src/ui/layout_eckhart/firmware/device_menu_screen.rs

```diff
@@ -0,0 +1,542 @@
+use core::ops::DerefMut;
+
+use crate::{
+    error::Error,
+    micropython::gc::GcBox,
+    strutil::TString,
+    trezorhal::storage::has_pin,
+    ui::{
+        component::{
+            text::{
+                paragraphs::{Paragraph, Paragraphs},
+                TextStyle,
+            },
+            Component, Event, EventCtx,
+        },
+        geometry::Rect,
+        layout_eckhart::{
+            component::{Button, ButtonStyleSheet},
+            constant::SCREEN,
+            firmware::{
+                Header, HeaderMsg, TextScreen, TextScreenMsg, VerticalMenu, VerticalMenuScreen,
+                VerticalMenuScreenMsg, SHORT_MENU_ITEMS,
+            },
+        },
+        shape::Renderer,
+    },
+};
+
+use super::{theme, ShortMenuVec};
+use heapless::Vec;
+
+const MAX_DEPTH: usize = 3;
+const MAX_SUBSCREENS: usize = 8;
+const MAX_SUBMENUS: usize = MAX_SUBSCREENS - 2 /* (about and device screen) */;
+
+const DISCONNECT_DEVICE_MENU_INDEX: usize = 1;
+
+#[derive(Clone)]
+enum Action {
+    // Go to another registered subscreen
+    GoTo(usize),
+
+    // Return a DeviceMenuMsg to the caller
+    Return(DeviceMenuMsg),
+}
+
+#[derive(Copy, Clone)]
+pub enum DeviceMenuMsg {
+    // Root menu
+    BackupFailed,
+
+    // "Pair & Connect"
+    DevicePair, // pair a new device
+    DeviceDisconnect(
+        usize, /* which device to disconnect, index in the list of devices */
+    ),
+
+    // Security menu
+    CheckBackup,
+    WipeDevice,
+
+    // Device menu
+    ScreenBrightness,
+    AutoLockDelay,
+
+    // nothing selected
+    Close,
+}
+
+struct MenuItem {
+    text: TString<'static>,
+    subtext: Option<(TString<'static>, Option<TextStyle>)>,
+    stylesheet: ButtonStyleSheet,
+    action: Option<Action>,
+}
+
+impl MenuItem {
+    pub fn new(text: TString<'static>, action: Option<Action>) -> Self {
+        Self {
+            text,
+            subtext: None,
+            stylesheet: theme::menu_item_title(),
+            action,
+        }
+    }
+
+    pub fn with_subtext(mut self, subtext: Option<(TString<'static>, Option<TextStyle>)>) -> Self {
```

Comment:
nit:
```suggestion
    pub fn with_subtext(&mut self, subtext: Option<(TString<'static>, Option<TextStyle>)>) -> &mut Self {
```
(also below)

Reply:
Right, should we do it also for other full-screen components, e.g. `TextScreen`?

Reply:
Yes, if possible (not urgent for this PR)

Reply:
Is there a difference? std seems to use [both](https://doc.rust-lang.org/std/fs/struct.DirBuilder.html#method.recursive) [styles](https://doc.rust-lang.org/std/thread/struct.Builder.html#method.stack_size) it seems ...

Reply:
IIUC, it's basically the same optimization as https://github.com/trezor/trezor-firmware/pull/4696

Reply:
We can do it for other components and builders but I'd not pollute this PR now.